dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/OpenAIChatMessageContent.cs

```diff
@@ -86,4 +83,41 @@ public IReadOnlyList<OpenAIFunctionToolCall> GetOpenAIFunctionToolCalls()
 
         return Array.Empty<OpenAIFunctionToolCall>();
     }
+
+    private static IReadOnlyDictionary<string, object?>? CreateMetadataDictionary(
+        IReadOnlyList<ChatCompletionsToolCall> toolCalls,
+        IReadOnlyDictionary<string, object?>? original)
+    {
+        // We only need to augment the metadata if there are any tool calls.
+        if (toolCalls?.Count > 0)
```

Comment:
The toolCalls parameter is not nullable, right?

Reply:
It is an `Optional<IReadOnlyList>`, not sure how this surface after the deserialization.

https://github.com/Azure/azure-sdk-for-net/blob/9a3cbe25fda60daeb7539707ad8d58dd19aa700a/sdk/openai/Azure.AI.OpenAI/src/Generated/ChatResponseMessage.Serialization.cs#L25C13-L25C61

Reply:
Addressed